### ISSUE TITLE
Wrong AppData folder

### ISSUE THREAD

**Issue by kobmik** [microsoft/vscode#162639]
- VS Code Version: 1.71.2
- OS Version: Windows 10 (Version 1909)

VS Code is using the wrong AppData folder with versions after 1.68.

With version 1.68, everything worked fine.
VS Code created AppData Files in the following folder:
\\server\usrprofile$\...\AppData (%APPDATA%)

With version, 1.71.2, VS Code is writing files to the following folder, which is wrong.
C:\usrprofile$\...\AppData

Has there anything changed regarding AppData path?

**Comment by benker44** [comment]
We have noticed the same problem. If a redirect of the %AppData% to a server is set up, certain plugins can no longer be loaded properly.
Example: Powershell Extension (v2022.8.5 on VSCode 1.72.1)
![Error_Popup_VSCode_20 10 2022_Terminal](https://user-images.githubusercontent.com/25610003/196933996-dc570ca6-00aa-4ee7-a5cc-e800209c6eab.png)
![Error_Popup_VSCode_20 10 2022](https://user-images.githubusercontent.com/25610003/196934033-351414f1-af42-476d-a4cd-cce849935480.png)

**Comment by lramos15** [comment]
Are you connected to a remote of some sort? I'm not sure what you mean by a redirect to a server?

**Comment by benker44** [comment]
Hi @lramos15, thanks for your comment
I use a VDI (Virtual Desktop Infrastructure) which is connected to our Active Directory and provided with profile synchronization and AppData Mirroring.

Here the Environment Info:
APPDATA                        \\wsenv01\usrprofile$\test\PRD\Win10x64\AppData

**Comment by bpasero** [comment]
I am not aware of changes on our end that would explain this. It also seems no other user has reported this so I suspect it could be a configuration issue on your side and not related to VS Code?

**Comment by kobmik** [comment]
It definitely has something to do with VS Code. With version 1.68 everything works well. With version 1.72, VS Code fails to access the AppData Folder.

If I switch to a local AppData Folder with the environment Variable "VSCODE_APPDATA" everything works fine. When I switch to a network share for the AppData Folder, VS Code automatically replaces the UNC path with "C:" and therefore VS Code fails to load the AppData.

**Comment by bpasero** [comment]
Maybe an Electron update or security fix, @deepak1556 any ideas?

**Comment by kobmik** [comment]
@bpasero : Is it possible to reopen the issue? It is still not fixed.

**Comment by kobmik** [comment]
Any news regarding this issue?

**Comment by bpasero** [comment]
No sorry for being slow here, you initially mentioned this folder was used: `\server\usrprofile$...\AppData`. What kind of setup is this where `\server` is being used? Some UNC path?

**Comment by kobmik** [comment]
Yes, AppData files/folders are not stored locally in our company.
The environment variable APPDATA is set to \\servername\usrprofile$\...\AppData
Somehow VS Code automatically replaces the server name path with C:\ and therefore VS Code can't find the files.

**Comment by bpasero** [comment]
You can always set the `--user-data-dir` argument to VS Code to change the user data dir. I think the issue here is that `%APPDATA%` used to be properly detected by VS Code but is not anymore?

**Comment by benker44** [comment]
I can specify the problem from my side. The error prevents me from starting certain plugins, e.g. Powershell. If I disable Powershell, VSCode works fine. Could the problem be with the plugins?

Electron isn't the problem either, unless they also offer plugin handling. Because I couldn't find any problems with GitHub Desktop and Postman.

**Comment by bpasero** [comment]
I would like an answer for https://github.com/microsoft/vscode/issues/162639#issuecomment-1330556205

**Comment by kobmik** [comment]
Same behaviour with `--user-data-dir \\servername\usrprofile$\...\AppData\Code`.
However `--user-data-dir c:\test\AppData\Code` works without any problems.
As soon I use a network share, VS Code doesn't work properly anymore.

**Comment by bpasero** [comment]
Sounds like an Electron change to me then, lets see what Deepak has to say.

**Comment by alexhass** [comment]
Redirected APPDATA folders (set with GPOs) are standard in enterprise environments.

This is a critical issue. Can you increase priority on this, please?

**Comment by deepak1556** [comment]
Seems like a side effect of the chromium sandbox, can you check if you are able to use the appdata from unc path if you run our [insiders](https://code.visualstudio.com/insiders) with `--disable-chromium-sandbox` flag.

**Comment by p3t3r5** [comment]
Tested with last VS Code Insiders with and without parameter `--disable-chromium-sandbox.` 
Folder created out of  redirected/networked %APPDATA% on  C:\user.name$\AppData\Code - Insiders\User\globalStorage
We have currently  trubles with redhat.java. extension. This is  extension only one aours extensions uses globalStorage and create some folders on open Java Project  in wrong globalStorage folder:
- redhat.java
- vscode-extension-recommender
- vscode-redhat-telemetry

Its VS Code or Extension Bug?
